main.go

```diff
@@ -1,272 +1,249 @@
+		GiveKarma:   karmaReg.GetGive(),
 		Leaderboard: regexp.MustCompile(`^karma(?:bot)? (?:leaderboard|top|highscores) ?([0-9]+)?$`),
 		URL:         regexp.MustCompile(`^karma(?:bot)? (?:url|web|link)?$`),
 		SlackUser:   regexp.MustCompile(`^<@([A-Za-z0-9]+)>$`),
 	}
-
-	debug                       bool
-	hasWebUI                    bool
-	hasMotivate                 bool
-	webUIURL                    string
-	maxPoints, leaderboardLimit int
-	bot                         *slack.Client
-	rtm                         *slack.RTM
-	ll                          *log.Logger
 )
 
-func main() {
-	ll = log.New(os.Stdout, "", log.Lshortfile|log.LstdFlags)
-	ll.Printf("karmabot v%s\n", VERSION)
-
-	var (
-		flagToken            = flag.String("token", "", "slack RTM token")
-		flagDBPath           = flag.String("db", "./db.sqlite3", "path to sqlite database")
-		flagMaxPoints        = flag.Int("maxpoints", 6, "the maximum amount of points that users can give/take at once")
-		flagLeaderboardLimit = flag.Int("leaderboardlimit", 10, "the default amount of users to list in the leaderboard")
-		flagDebug            = flag.Bool("debug", false, "set debug mode")
-		flagTOTP             = flag.String("totp", "", "totp key")
-		flagWebUIPath        = flag.String("webuipath", "", "path to web UI files")
-		flagListenAddr       = flag.String("listenaddr", "", "address to listen and serve the web ui on")
-		flagWebUIURL         = flag.String("webuiurl", "", "url address for accessing the web ui")
-		flagMotivate         = flag.Bool("motivate", true, "toggle motivate.im support")
-	)
-
-	flag.Parse()
-	maxPoints = *flagMaxPoints
-	leaderboardLimit = *flagLeaderboardLimit
-	debug = *flagDebug
-	hasWebUI = *flagWebUIPath != "" && *flagListenAddr != ""
-	hasMotivate = *flagMotivate
-	if hasWebUI {
-		if *flagWebUIURL != "" {
-			webUIURL = *flagWebUIURL
-		} else {
-			webUIURL = fmt.Sprintf("http://%s/", *flagListenAddr)
-		}
-	}
-
-	if *flagToken == "" {
-		ll.Fatalln("please pass the slack RTM token using the -token option")
-	}
-
-	database.Init(ll, *flagDBPath)
-
-	bot = slack.New(*flagToken)
-	slack.SetLogger(ll)
-	bot.SetDebug(debug)
+type Slack struct {
+	Bot *slack.Client
+	RTM *slack.RTM
+}
 
-	rtm = bot.NewRTM()
-	go rtm.ManageConnection()
+type Config struct {
+	Slack                       *Slack
+	Debug, Motivate             bool
+	MaxPoints, LeaderboardLimit int
+	Log                         *log.Log
+	UI                          ui.Provider
+	DB                          *database.DB
+}
 
-	if hasWebUI {
-		webUIConfig := &webui.Config{
-			Logger:           ll,
-			TOTPKey:          *flagTOTP,
-			ListenAddr:       *flagListenAddr,
-			FilesPath:        *flagWebUIPath,
-			LeaderboardLimit: leaderboardLimit,
-			Debug:            debug,
-		}
+type Bot struct {
+	Config *Config
+}
 
-		webui.Init(webUIConfig)
-		go webui.Listen()
+func New(config *Config) *Bot {
+	return &Bot{
+		Config: config,
```

Comment:
This function doesn't do anything beyond just using a struct literal, so why have it? It won't even prevent people from changing the config on you once they've created a `Bot` since it's a pointer :)